update pebble dependency of TiDB repo

## Enhancement

when tiflow repo wants to update tidb library version, because they are using two different version of pebble and API is not compatible, tiflow can't build

```
../../../../pkg/mod/github.com/pingcap/tidb@v1.1.0-beta.0.20230823042435-8d0fdef09869/util/extsort/disk_sorter.go:347:36: not enough arguments in call to si.iter.SeekGE
        have (nil)
        want ([]byte, bool)
```

https://github.com/pingcap/tiflow/blob/e5c78518c0d7517242aa6bd9fabc6da484e53aa4/go.mod#L20

https://github.com/pingcap/tidb/blob/8d0fdef09869a43ff8d2f14749f7afc38bf56514/go.mod#L29

--- a/DEPS.bzl
+++ b/DEPS.bzl
@@ -1139,13 +1139,13 @@ def go_deps():
         name = "com_github_cockroachdb_pebble",
         build_file_proto_mode = "disable_global",
         importpath = "github.com/cockroachdb/pebble",
-        sha256 = "06d2cb18472b270a8c41a14b5c23c3e52065d54ee25b0c4d7aa55a63dfb45470",
-        strip_prefix = "github.com/cockroachdb/pebble@v0.0.0-20210719141320-8c3bd06debb5",
+        sha256 = "6ecda6200019988f0453986e2f736fe588b94f62bb6db6cdc9f8ba783261d750",
+        strip_prefix = "github.com/cockroachdb/pebble@v0.0.0-20220415182917-06c9d3be25b3",
         urls = [
-            "http://bazel-cache.pingcap.net:8080/gomod/github.com/cockroachdb/pebble/com_github_cockroachdb_pebble-v0.0.0-20210719141320-8c3bd06debb5.zip",
-            "http://ats.apps.svc/gomod/github.com/cockroachdb/pebble/com_github_cockroachdb_pebble-v0.0.0-20210719141320-8c3bd06debb5.zip",
-            "https://cache.hawkingrei.com/gomod/github.com/cockroachdb/pebble/com_github_cockroachdb_pebble-v0.0.0-20210719141320-8c3bd06debb5.zip",
-            "https://storage.googleapis.com/pingcapmirror/gomod/github.com/cockroachdb/pebble/com_github_cockroachdb_pebble-v0.0.0-20210719141320-8c3bd06debb5.zip",
+            "http://bazel-cache.pingcap.net:8080/gomod/github.com/cockroachdb/pebble/com_github_cockroachdb_pebble-v0.0.0-20220415182917-06c9d3be25b3.zip",
+            "http://ats.apps.svc/gomod/github.com/cockroachdb/pebble/com_github_cockroachdb_pebble-v0.0.0-20220415182917-06c9d3be25b3.zip",
+            "https://cache.hawkingrei.com/gomod/github.com/cockroachdb/pebble/com_github_cockroachdb_pebble-v0.0.0-20220415182917-06c9d3be25b3.zip",
+            "https://storage.googleapis.com/pingcapmirror/gomod/github.com/cockroachdb/pebble/com_github_cockroachdb_pebble-v0.0.0-20220415182917-06c9d3be25b3.zip",
         ],
     )
     go_repository(

--- a/br/pkg/lightning/backend/local/local_test.go
+++ b/br/pkg/lightning/backend/local/local_test.go
@@ -175,7 +175,7 @@ func TestRangeProperties(t *testing.T) {
 	for _, p := range cases {
 		v := make([]byte, p.vLen)
 		for i := 0; i < p.count; i++ {
-			_ = collector.Add(pebble.InternalKey{UserKey: p.key, Trailer: pebble.InternalKeyKindSet}, v)
+			_ = collector.Add(pebble.InternalKey{UserKey: p.key, Trailer: uint64(pebble.InternalKeyKindSet)}, v)
 		}
 	}
 
@@ -291,7 +291,7 @@ func TestRangePropertiesWithPebble(t *testing.T) {
 			binary.BigEndian.PutUint64(key, uint64(i*100+j))
 			err := wb.Set(key, value[:valueLen], writeOpt)
 			require.NoError(t, err)
-			err = collector.Add(pebble.InternalKey{UserKey: key, Trailer: pebble.InternalKeyKindSet}, value[:valueLen])
+			err = collector.Add(pebble.InternalKey{UserKey: key, Trailer: uint64(pebble.InternalKeyKindSet)}, value[:valueLen])
 			require.NoError(t, err)
 		}
 		require.NoError(t, wb.Commit(writeOpt))

--- a/go.mod
+++ b/go.mod
@@ -26,7 +26,7 @@ require (
 	github.com/cheynewallace/tabby v1.1.1
 	github.com/cloudfoundry/gosigar v1.3.6
 	github.com/cockroachdb/errors v1.8.1
-	github.com/cockroachdb/pebble v0.0.0-20210719141320-8c3bd06debb5
+	github.com/cockroachdb/pebble v0.0.0-20220415182917-06c9d3be25b3
 	github.com/coocood/freecache v1.2.1
 	github.com/coreos/go-semver v0.3.0
 	github.com/cznic/mathutil v0.0.0-20181122101859-297441e03548

--- a/util/extsort/disk_sorter.go
+++ b/util/extsort/disk_sorter.go
@@ -210,10 +210,8 @@ func (sw *sstWriter) Close() (retErr error) {
 		meta := &fileMetadata{
 			fileNum: sw.fileNum,
 		}
-		smallest := writerMeta.Smallest(pebble.DefaultComparer.Compare)
-		largest := writerMeta.Largest(pebble.DefaultComparer.Compare)
-		meta.startKey = slices.Clone(smallest.UserKey)
-		meta.lastKey = slices.Clone(largest.UserKey)
+		meta.startKey = slices.Clone(writerMeta.SmallestPoint.UserKey)
+		meta.lastKey = slices.Clone(writerMeta.LargestPoint.UserKey)
 		// Make endKey is exclusive. To avoid unnecessary overlap,
 		// we append 0 to make endKey is the smallest key which is
 		// greater than the last key.
@@ -339,12 +337,12 @@ type sstIter struct {
 }
 
 func (si *sstIter) Seek(key []byte) bool {
-	si.key, si.value = si.iter.SeekGE(key)
+	si.key, si.value = si.iter.SeekGE(key, false)
 	return si.key != nil
 }
 
 func (si *sstIter) First() bool {
-	si.key, si.value = si.iter.SeekGE(nil)
+	si.key, si.value = si.iter.SeekGE(nil, false)
 	return si.key != nil
 }
 

